bundles/automation/org.eclipse.smarthome.automation.module.timer/src/main/java/org/eclipse/smarthome/automation/module/timer/handler/TimeOfDayConditionHandler.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.automation.module.timer.handler;
+
+import java.time.LocalTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Map;
+
+import org.eclipse.smarthome.automation.Condition;
+import org.eclipse.smarthome.automation.handler.BaseModuleHandler;
+import org.eclipse.smarthome.automation.handler.ConditionHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * ConditionHandler implementation for time based conditions.
+ *
+ * @author Dominik Schlierf - initial contribution
+ *
+ */
+public class TimeOfDayConditionHandler extends BaseModuleHandler<Condition> implements ConditionHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(TimeOfDayConditionHandler.class);
+
+    public static final String MODULE_TYPE_ID = "core.TimeOfDayCondition";
+
+    /**
+     * Constants for Config-Parameters corresponding to Definition in
+     * TimeOfDayConditionHandler.json
+     */
+    private static final String START_TIME = "startTime";
+    private static final String END_TIME = "endTime";
+
+    public TimeOfDayConditionHandler(Condition condition) {
+        super(condition);
+    }
+
+    @Override
+    public boolean isSatisfied(Map<String, Object> inputs) {
+
+        String startTimeConfig = (String) module.getConfiguration().get(START_TIME);
+        String endTimeConfig = (String) module.getConfiguration().get(END_TIME);
+        if (startTimeConfig == null || endTimeConfig == null) {
+            logger.error("Time condition with id {} is not well configured: startTime={}  endTime = {}", module.getId(),
```

Comment:
The *error* log level is used for serious events only. A user configuration mistake is maximum a warning.

Reply:
Ok!